bundle/regal/rules/imports/unresolved-reference/unresolved_reference.rego

```diff
@@ -0,0 +1,122 @@
+# METADATA
+# description: Reference to unresolved field.
+package regal.rules.imports["unresolved-reference"]
+
+import data.regal.ast
+import data.regal.config
+import data.regal.result
+import data.regal.util
+
+# METADATA
+# description: collects exported and full of used refs from each module
+aggregate contains content if {
+	# regal ignore:unconditional-assignment
+	content := result.aggregate(rego.metadata.chain(), {
+		"exported_rules": object.keys(ast.rule_head_locations),
+		"expanded_refs": _all_full_path_refs,
+		"prefix_tree": _prefix_tree,
+	})
+}
+
+_import_aliases[alias_key] := string_value if {
+	some alias_key, value in ast.resolved_imports
+	string_value := concat(".", value)
+}
+
+# an import is shadowed if it shares name with a rule
+_shadowed_imports contains rule_name if {
+	some rule_name in ast.rule_names
+	_import_aliases[rule_name]
+}
+
+# an import is shadowed if it shares name with a variable (or function argument)
+_shadowed_imports contains var_name if {
+	var_name := ast.found.vars[_][_][_].value
+	_import_aliases[var_name]
+}
+
+_refs contains ref if {
+	_ref := ast.found.refs[_][_].value
```

Comment:
No need to use underscore prefixes for local vars. You could call this `terms` if you want to use `ref` later.

`_name` could just be called `name`